src/trio/_core/_tests/test_run.py

```diff
@@ -2379,6 +2384,8 @@ async def test_cancel_scope_deadline_duplicates() -> None:
         await sleep(0.01)
 
 
+# I don't know if this one can fail anymore, the `del` next to the comment that used to
+# refer to this only seems to break test_cancel_scope_exit_doesnt_create_cyclic_garbage
```

Comment:
@richardsheridan said in https://github.com/python-trio/trio/pull/2891#pullrequestreview-1785190814
> Hanging on to `test_simple_cancel_scope_usage_doesnt_create_cyclic_garbage` even though it maybe can't fail currently is fine IMO since we want to cover outcome and protect future refactoring.

I'd probably weakly be in favor of removing the test, but if not should probably update the comment to include sheridan's reasoning.

though `concat_tb` is only used for `collapse_exception_group`, which is only used for `strict_exception_groups=False`, which we might deprecate soonish. So not sure there's much future refactoring to consider.

Reply:
Yeah updating the comment would be nice

Reply:
Ah oops, I mixed up the tests - the latter half of my previous comment pertains to `test_ExceptionGroup_catch_doesnt_create_cyclic_garbage`.
Updated the comment though.